### ISSUE TITLE
Is HTML Code_Sniffer Enough for Accessibility Testing ? 

### ISSUE THREAD

**Issue by reach2akram** [squizlabs/HTML_CodeSniffer#114]
I am using HTML Code_Sniffer for my web application and testing till WCAG 2.0 AA. I am covering All the errors and warning.

Could you please suggest that after fixing all the errors and warnings till WCAG 2.0 AA on my web application using HTML Code_Sniffer , Will it be completely WCAG 2.0 AA compliance  ?

**Comment by gsherwood** [comment]
> Will it be completely WCAG 2.0 AA compliance

That is impossible to say because some checks must be done manually and are shown as notices. If you want to guarantee that your site is accessible, you'll need to get a professional audit done by an accessibility expert.

But fixing all the errors and warnings will often get you full compliance, or very close to it.

It is important to understand that none of the automated tools you use for checking accessibility can give you a certification. You need to get those manually.

**Comment by pgindia** [comment]
that's really a helpful information. What about screen readers and keyboard accessibility testing. Does HTML code sniffer pick errors related to them - ARIA label, tab index etc.

**Comment by lwright-sq** [comment]
Hi there,

Support for ARIA labelling attributes - ie. aria-label and aria-labelledby - was [added recently as an supported option for labelling](http://squizlabs.github.io/HTML_CodeSniffer/Patches/2.0.2), which follows the expansion of the suggested options in the appropriate technique.

Tabindexes are only provided as a generic "notice" message, requesting that you manually check that the tabindex values follow the natural flow of the form. This is because HTMLCS cannot automatically determine whether the order of fields in HTML match the order they appear on the screen, or assume anything about what the tab order should be. For instance, if you style your form with two columns of text fields using side-by-side DIVs or similar, the natural tabbing flow would probably go left-to-right before going top-to-bottom, but they won't appear that way in the HTML.
(Whether such things would violate other rules about being understandable when linearised is another matter - it would still be understandable as a form but the order of fields may not make sense. Again though, this sounds like something that needs manual thought to decide.)

**Comment by pgindia** [comment]
Thanks for the reply, that was really helpful.
